### PR TITLE
Better use of WORKDIR in dockerfile

### PR DESCRIPTION
**Pull request by kmehant**
Signed-off-by: K mehant <411843@student.nitandhra.ac.in>

### CODE REVIEW

**Comment by kmehant** [comment]
@sykesm Please review.

File: .gitignore [hunk 1]
```diff
@@ -7,3 +7,4 @@
 .*.sw*
 /build
 .idea
+.vscode
```
Comment (btl5037) [inline .gitignore:10]: You accidentally removed the newline at the end of the end of the file

**Comment by btl5037** [comment]
I don't mind the explicit creation of the tools directory, as it shows it didn't already exist. Though I have no opinion either way.

It's probably more idiomatic to be explicit in the fact we changed into /tools/proto rather than making the directive relative as you've done in this change. 

Matt, do you want me to setup CI to move this to the Hyperledger Artifactory and create a standalone CI images repo and migrate this over there. Then we can automate the publishing of these images when they are updated

**Comment by kmehant** [comment]
> I don't mind the explicit creation of the tools directory, as it shows it didn't already exist. Though I have no opinion either way.
> 
> It's probably more idiomatic to be explicit in the fact we changed into /tools/proto rather than making the directive relative as you've done in this change.
> 
> Matt, do you want me to setup CI to move this to the Hyperledger Artifactory and create a standalone CI images repo and migrate this over there. Then we can automate the publishing of these images when they are updated

Thanks for the comments, I wish to work on the CI (trigger -> lint -> build -> push to hub) if you are having plans in bringing up a separate repo for all container configurations :)

File: .gitignore (hunk 1 above)
Comment (kmehant) [reply .gitignore]: sorry, fixed :)

**Comment by sykesm** [comment]
> Matt, do you want me to setup CI to move this to the Hyperledger Artifactory and create a standalone CI images repo and migrate this over there. Then we can automate the publishing of these images when they are updated

CI images repo? No. I still assert that having everything in the ci folder of the repo is a lot easer to reason about than spreading it all over with a bunch of automation. I've lived in the land of the "project-dockerfiles" repo and decided to relocate to a nicer place. At the end of the day, it just adds more steps to updating and leaves more garbage lying around.

Artifactory? Maybe. Github Packages or Docker Hub? Sure, with a manual trigger. Both can be automated within the context of the consuming repo, though, and neither should require a "release" to push.

As for the change, it's non-functional so I'm fine with merging. Before doing that, the commits should be squashed. I'll add that since it is non-functional, there's no need to build and publish the image at this time.

**Comment by kmehant** [comment]
@sykesm I have squashed the commits.

**Pull request merged**